app.py

```diff
@@ -60,164 +87,289 @@ def extract_graph_id(fx_graph_lines):
+                            post_grad_node_to_lines[post_grad_node_name]
+                        )
+    
+    # Process postToCppCode using lookup maps
+    for post_grad_node, cpp_code_kernel_names in node_mappings["postToCppCode"].items():
+        if post_grad_node in post_grad_node_to_lines:
+            gen_line_num = post_grad_node_to_lines[post_grad_node]
+            line_post_to_cpp_code[gen_line_num] = []
+            for cpp_code_kernel_name in cpp_code_kernel_names:
+                if cpp_code_kernel_name in cpp_code_to_lines:
+                    line_post_to_cpp_code[gen_line_num].extend(
+                        cpp_code_to_lines[cpp_code_kernel_name]
+                    )
+
+    return {"preToPost": line_pre_to_post, "postToPre": line_post_to_pre, "pyCodeToPost": line_py_code_to_post, "postToPyCode": line_post_to_py_code,
+            "cppCodeToPost": line_cpp_code_to_post, "postToCppCode": line_post_to_cpp_code}
 
-    return {
-        'leftToRight': line_left_to_right,
-        'rightToLeft': line_right_to_left
-    }
 
-@app.route('/')
+@app.route("/")
 def index():
-    return render_template('index.html')
+    return render_template("index.html")
 
-@app.route('/load_file', methods=['POST'])
+
+@app.route("/load_file", methods=["POST"])
 def load_file():
     """Endpoint to load file content from URL"""
     try:
         data = request.json
-        url = data.get('url')
+        url = data.get("url")
         if not url:
-            return jsonify({'error': 'No URL provided'}), 400
-        
+            return jsonify({"error": "No URL provided"}), 400
+
         lines = fetch_content_from_url(url)
-        return jsonify({'lines': lines})
-        
+        return jsonify({"lines": lines})
+
     except requests.RequestException as e:
-        return jsonify({'error': f'Failed to fetch URL: {str(e)}'}), 400
+        return jsonify({"error": f"Failed to fetch URL: {str(e)}"}), 400
     except Exception as e:
-        return jsonify({'error': f'Unexpected error: {str(e)}'}), 500
+        return jsonify({"error": f"Unexpected error: {str(e)}"}), 500
+
 
-@app.route('/process_mapping', methods=['POST'])
+@app.route("/process_mapping", methods=["POST"])
 def process_mapping():
     """Endpoint to process mappings between FX graph and generated code"""
     try:
         data = request.json
-        fx_graph_lines = data['fxGraphData']
-        post_grad_graph_lines = data['codeData']
-        
+        fx_graph_lines = data["fxGraphData"]
+        post_grad_graph_lines = data["postGradGraphData"]
+        python_code_lines = data["codeData"]
+        cpp_code_lines = data["cppCodeData"]
+
+        # Read the local inductor triton kernel to post grad nodes mapping
+        # TODO: this path is hardcoded for now, need to make it configurable
+        # maybe consolidate with the other json parsing for pre_grad to post_grad
+        file_path = 'tl_out/inductor_triton_kernel_to_post_grad_nodes.json'
```

Comment:
@YUNQIUGUO fyi, for now, my json is just a comment at the end of the post_grad_graph file. But maybe it's a better idea to store it as a separate file? I wonder if we can let tlparse consolidate all json files and all 4 graphs files into a single file, so the user only needs to load a single file for all panels and mappings.

Reply:
> I wonder if we can let tlparse consolidate all json files and all 4 graphs files into a single file, so the user only needs to load a single file for all panels and mappings.

yeah, agree. that's my thought to consolidate too. 
right now my json file is stored and dumped to model `debug_trace` folder when TORCH_COMPILE_DEBUG flag is enabled.
let's take a look at the idea of storing it as one separate file.